### PR TITLE
Fixing incorrect type on canMakeNativePayPayments

### PR DESCRIPTION
**Pull request by alexstuckey**
`.canMakeNativePayPayments([options])` allows a list of 'networks' to be passed to the Apply Pay equivalent method.
The Tipsi documentation lists this as `[options]`, suggesting a list; however, it specifically references the key 'networks' as an option to fill, requiring a map for implementation.

This led to an `-[NSObject(NSObject) doesNotRecognizeSelector:]` error on iOS, as an List was passed, which was then queried for a value by a key – an illegal operation for an NSArrayM.

This patch fixes this by making `canMakeNativePayPayments`'s argument a Map called 'options'.

### CODE REVIEW

File: lib/src/stripe_payment.dart [hunk 1]
```diff
@@ -33,11 +33,11 @@ class StripePayment {
   }
 
   /// https://tipsi.github.io/tipsi-stripe/docs/canMakeNativePayPayments.html
-  static Future<bool> canMakeNativePayPayments(List<String> networks) async {
+  static Future<bool> canMakeNativePayPayments(Map<String, dynamic> options) async {
```
Comment (jonasbark) [inline lib/src/stripe_payment.dart]: Could you make this a bit more type-safe?
E.g. keep the `List<String> networks` parameter and send over a map instead, just like below:
`{"networks": networks}`

Comment (alexstuckey) [reply lib/src/stripe_payment.dart]: My rationale for `Map<String, dynamic> options` was to mirror the Tipsi project as much as possible.
Their current implementation is here https://github.com/tipsi/tipsi-stripe/blob/1d89c82ac8d5c607bd5563c19ac85a7854be39aa/src/Stripe.js#L66

Comment (jonasbark) [reply lib/src/stripe_payment.dart]: I agree on that but then again it's javascript - all other methods aren't typed as well and that's not really something we'd want in a Dart library :-)

Comment (jonasbark) [reply lib/src/stripe_payment.dart]: Thanks for the fix! Could you change the key to "networks"? That should be the right one.

Comment (alexstuckey) [reply lib/src/stripe_payment.dart]: Ah! Apologies for the typo

**Pull request merged**